v6.0 Beta to GA Updates - Config Settings (#4983)
mattermost/docs: source/onboard/ad-ldap.md

If this filter is removed/changed, active guests will not be promoted to a member and will retain their Guest role. Guests can be promoted in **System Console > User Management**.

1. Navigate to **System Console > Authentication > Guest Access** and set Guest Access to `true`.
2. Navigate to **System Console > Authentication > AD/LDAP**.
3. Complete the **Guest Filter** field.
4. Choose **Save**.

> When a guest logs in for the first time they are presented with a default landing page until they are added to channels.

Reviewer: ```suggestion
When a guest logs in for the first time they are presented with a default landing page until they're added to channels.
```

There are three AD/LDAP attributes that apear to be similar but serve a different purpose:

1. **Username Attribute:** Used within the Mattermost user interface to identify and mention users. For example, if **Username Attribute** is set to `john.smith`, a user typing `@john` will see `@john.smith` in their autocomplete options and posting a message with `@john.smith` will send a notification to that user that they’ve been mentioned.

> 2. **ID Attribute:** Used as the unique identifier in Mattermost. It should be an AD/LDAP attribute with a value that does not change, such as `ObjectGUID`. If a user's ID attribute changes, it will create a new Mattermost account unassociated with their old one. If you need to change this field after users have already logged in, use the [mattermost ldap idmigrate CLI tool](https://docs.mattermost.com/manage/command-line-tools.html#mattermost-ldap-idmigrate).

Reviewer: ```suggestion
2. **ID Attribute:** Used as the unique identifier in Mattermost. It should be an AD/LDAP attribute with a value that doesn't change, such as `ObjectGUID`. If a user's ID attribute changes, it will create a new Mattermost account unassociated with their old one. If you need to change this field after users have already logged in, use the [mattermost ldap idmigrate CLI tool](https://docs.mattermost.com/manage/command-line-tools.html#mattermost-ldap-idmigrate).
```

There are two main ways to do this:

1. **User deletion:** If the user is completely removed from the AD/LDAP server, they will be deactivated in Mattermost on the next synchronization.

> 2. **User filter:** Set the [user filter](https://docs.mattermost.com/configure/configuration-settings.html#user-filter) to only select the subset of AD/LDAP users you want to have access to Mattermost. When someone is removed from the selected group, they will be deactivated in Mattermost on the next synchronization.

Reviewer: ```suggestion
2. **User filter:** Set the [user filter](https://docs.mattermost.com/configure/configuration-settings.html#user-filter) to only select the subset of AD/LDAP users you want to have access to Mattermost. When someone is removed from the selected group, they'll be deactivated in Mattermost on the next synchronization.
```

This means the query sent back to the AD/LDAP server returned no results. We recommend that you:

> 1. Check that the user credentials were entered properly - you should log in with the field set as the [`ID Attribute`](https://docs.mattermost.com/configure/configuration-settings.html#id-attribute).

Reviewer: ```suggestion
1. Check that the user credentials were entered correctly - you should log in with the field set as the [`ID Attribute`](https://docs.mattermost.com/configure/configuration-settings.html#id-attribute).
```

To do this, you can set the [**Login ID Attribute**](https://docs.mattermost.com/configure/configuration-settings.html#login-id-attribute) to whatever you would like users to log in with (e.g. username or email).

**Note:** Currently the value is case sensitive. If the **ID Attribute** is set to the username and the username changes from `John.Smith` to `john.smith`, the user will experience problems logging in.

### I see the log error `LDAP Result Code 4 "Size Limit Exceeded"`.

This indicates your AD/LDAP server configuration has a maximum page size set and the query coming from Mattermost is returning a result set in excess of that limit.

To address this issue you can set the [max page size](https://docs.mattermost.com/configure/configuration-settings.html#maximum-page-size) in your Mattermost configuration to match the limit on your AD/LDAP server. This will return a sequence of result sets that do not exceed the max page size, rather than returning all results in a single query. A max page size setting of 1500 is recommended.

> If the error is still occurring, it is likely that no AD/LDAP users have logged into Mattermost yet. Ensure that at least one AD/LDAP user has logged into Mattermost and re-run the sync. The error should disappear at that point.

Reviewer: ```suggestion
If the error is still occurring, it's likely that no AD/LDAP users have logged into Mattermost yet. Ensure that at least one AD/LDAP user has logged into Mattermost and re-run the sync. The error should disappear at that point.
```
